extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -667,7 +675,8 @@ export class AmpConsent extends AMP.BaseElement {
         this.updateCacheIfNotNull_(
           response['consentStateValue'],
           response['consentString'] || undefined,
-          response['consentMetadata'] || undefined
+          response['consentMetadata'] || undefined,
+          response['purposeConsents'] || undefined
```

Comment:
If a property is not found in an object, it will return `undefined`, so can remove `|| undefined`.

Reply:
The consumer type might need a change:

```diff
  /**
   * Sync with local storage if consentRequired is true.
   *
-  * @param {string=} responseStateValue
+  * @param {?string=} responseStateValue
-  * @param {string=} responseConsentString
+  * @param {?string=} responseConsentString
   * @param {JsonObject=} opt_responseMetadata
   */
  updateCacheIfNotNull_(
  ```